lib/STM_domain.ml

```diff
@@ -34,6 +38,29 @@ module Make (Spec: Spec) = struct
            (fun (c,r) -> Printf.sprintf "%s : %s" (Spec.show_cmd c) (show_res r))
            (pref_obs,obs1,obs2)
 
+  let agree_prop_par_asym (seq_pref, cmds1, cmds2) =
+    let sut = Spec.init_sut () in
+    let pref_obs = interp_sut_res sut seq_pref in
+    let sema = Semaphore.Binary.make false in
```

Comment:
Is there a reason to use a `Semaphore` rather than an `Atomic` like in the symmetric version?